src/net_processing.cpp

```diff
@@ -2141,17 +2141,15 @@ void PeerManagerImpl::ProcessGetBlockData(CNode& pfrom, Peer& peer, const CInv&
         // Fast-path: in this case it is possible to serve the block directly from disk,
         // as the network format matches the format on disk
         std::vector<uint8_t> block_data;
-        if (!ReadRawBlockFromDisk(block_data, pindex->GetBlockPos(), m_chainparams.MessageStart())) {
-            assert(!"cannot load block from disk");
-        }
+        // fail if cannot load block from disk
+        assert(ReadRawBlockFromDisk(block_data, pindex->GetBlockPos(), m_chainparams.MessageStart()));
```

Comment:
Could you please split this up like this instead?
```c++
auto foo = ReadRawBlockFromDisk(...);
assert(foo);
```
That way the we don't introduce a new side-effect of `NDEBUG=1`